i18n(pt-BR): Adding `imports.md` and updating `astro-components.md` (#532)
withastro/docs: src/pages/pt-BR/guides/imports.md

import json from './dados.json';
```

Astro dá suporte para a importação de arquivos JSON diretamente em sua aplicação. Arquivos importados retornam o objeto JSON completo no `import` padrão.

## CSS

```js
// Carrega e injeta o arquivo 'estilos.css' na página
import './estilos.css';
```

Astro suporta a importação de arquivos CSS diretamente em sua aplicação. Estilos importados não proveem nenhum "export", mas importar um arquivo CSS irá automaticamente adicionar seus estilos à página. Isto funciona para todos os arquivos CSS por padrão e pode suportar também pré-processadores como Sass e Less via plugins.

Se você prefere não escrever CSS, Astro também suporta todas as bibliotecas populares de CSS-in-JS (ex: styled-components) para estilização.

## CSS Modules

```jsx
// 1. Converte os nomes das classes de `./estilos.module.css` para valores únicos e escopados.
// 2. Retorna um objeto que mapeia os nomes das classes originais aos seus valores únicos e escopados.
import estilos from './estilo.module.css';
// Esse exemplo usa JSX, mas você pode usar Módulos CSS com qualquer framework.
return <div className={estilos.erro}>Sua Mensagem de Erro</div>;
```

> Astro da o suporte para CSS Modules usando a convenção de nome `[nome].module.css`. Como qualquer arquivo CSS, importá-lo vai automaticamente aplicar os estilos à página. Entretando, CSS Modules exportam um objeto padão de estilo que mapeia os nomes originais das classes à seus identificadores únicos.

Author: ```suggestion
Astro suporta CSS Modules usando a convenção de nome `[nome].module.css`. Como qualquer arquivo CSS, importá-lo vai automaticamente aplicar os estilos à página. Entretando, CSS Modules exportam um objeto padrão que mapeia os nomes originais das suas classes em identificadores únicos.
```

Os tipos de arquivos abaixo são suportados pelo Astro sem configuração necessária:

- Astro Components (`.astro`)
- Markdown (`.md`)
- JavaScript (`.js`, `.mjs`)
- TypeScript (`.ts`, `.tsx`)
- NPM Packages
- JSON (`.json`)
- JSX (`.jsx`, `.tsx`)
- CSS (`.css`)
- CSS Modules (`.module.css`)
- Imagens e Assets (`.svg`, `.jpg`, `.png`, etc.)

Se você não viu o tipo de asset que está procurando, veja a nossa [Biblioteca de Integrações](https://astro.build/integrations/). Você pode extender Astro para adicionar o suporte para diferente tipos de arquivos, como componentes Svelte e Vue.

Este guia detalha como os diferentes tipos de assets são construídos pelo Astro e como importá-los corretamente.

Lembre-se que você pode colocar qualquer asset estático no [diretório `public/`](/pt-BR/core-concepts/project-structure/#public) do seu projeto e Astro irá copiá-los diretamente para seu código final. Arquivos do diretório `public/` não são construídos ou empacotado por Astro, o que signifca que qualquer tipo de arquivo é suportado. Você pode referenciar um arquivo do diretório `public/` por um caminho de URL diretamente em seu template HTML.

## JavaScript

```js

> import { getUser } from './user.js';

Author: I also tought about doing this, I agree. Can you just edit your suggestion? There's a typo `usario.js` -> `usuario.js`

> import { getUser } from './user.js';

Reviewer: Fixed it!

> import { getUser } from './user.js';

Author: ```suggestion
import { getUsuario } from './usuario.js';
```

```

Astro inclui suporte padrão para arquivos JSX (`*.jsx` and `*.tsx`) em seu projeto. A sintaxe JSX é automaticamente transpilada para JavaScript.

Apesar do Astro entender a sintaxe JSX por padrão, você deverá incluir a integração de framework UI adequada para renderizar componentes React, Preact e Solid. Confira o nosso guia [Usando Integrações](/pt-BR/guides/integrations-guide) para saber mais.

**Nota: Astro não da suporte à JSX em arquivos `.js`/`.ts`**. JSX será manipulado apenas dentro de arquivo que terminam com as extensões `.jsx` ou `.tsx`.

## Pacotes NPM

```js
// Importa os pacotes npm React e React-DOM
import React from 'react';
import ReactDOM from 'react-dom';
```

Astro permite que você importe pacotes NPM diretamente no navegador. Mesmo se um pacote foi publicado usando um formato legado, Astro irá convertê-lo para ESM antes de serví-lo ao navegador.

## JSON

```js
// Carrega o objeto JSON pelo "export" padrão.
import json from './data.json';
```

Astro da o suporte para a importação de arquivos JSON diretamente em sua aplicação. Arquivos importados retornam o objeto JSON completo no `import` padrão.

## CSS

```js

> // Carrega e injeta o arquivo 'style.css' na página
> import './style.css';

Reviewer: ```suggestion
// Carrega e injeta o arquivo 'estilos.css' na página
import './estilos.css';
```

```

> Astro da o suporte para a importação de arquivos CSS diretamente em sua aplicação. Estilos importados não proveêm nenhum "export", mas importar um arquivo CSS automaticamente adiciona este estilo à página. Isto funciona para todos os arquivos CSS por padrão e pode suportar também os pré-processadores como Sass e Less via plugins.

Reviewer: ```suggestion
Astro suporta a importação de arquivos CSS diretamente em sua aplicação. Estilos importados não proveem nenhum "export", mas importar um arquivo CSS irá automaticamente adicionar seus estilos à página. Isto funciona para todos os arquivos CSS por padrão e pode suportar também pré-processadores como Sass e Less via plugins.
```

> Se você prefere não escrever CSS, Astro também suporta todas as bibliotecas populares de CSS-in-JS (exemplo: styles-components) para estilização.

Reviewer: ```suggestion
Se você prefere não escrever CSS, Astro também suporta todas as bibliotecas populares de CSS-in-JS (ex: styled-components) para estilização.
```

> ## CSS Modules

Reviewer: ```suggestion
## Módulos CSS
```

Author: I don't know if we should translate this, I think this it's a common keyword like React, Vue and etc.

Reviewer: My reasoning for translating CSS Modules, is that, AFAIK, CSS Modules are like a spec for different build and processing tools to implement, although not an official one. For example, `Selectors Level 4` are translated to `Seletores Nível 4` on a lot of websites I found. I don't really have strong opinions on this, feel free to decide.

Author: That's alright, I'll keep the translation! If we chang our mind, we can update this later.

```jsx

> // 1. Converte os nomes das classes de `./style.module.css` para valores únicos e escopados.
> // 2. Retorn um objecto que mapeia os nomes das classes originais aos valores únicos e escopados.
> import styles from './style.module.css';
> 
> // Esse exemplo usa JSX, mas você pode usar CSS Modules com qualquer framework.
> return <div className={styles.error}>Sua Mensagem de Erro</div>;

Reviewer: ```suggestion
// 1. Converte os nomes das classes de `./estilos.module.css` para valores únicos e escopados.
// 2. Retorna um objeto que mapeia os nomes das classes originais aos seus valores únicos e escopados.
import estilos from './estilo.module.css';
// Esse exemplo usa JSX, mas você pode usar Módulos CSS com qualquer framework.
return <div className={estilos.erro}>Sua Mensagem de Erro</div>;
```

```

> Astro da o suporte para CSS Modules usando a convenção de nome `[nome].module.css`. Como qualquer arquivo CSS, importá-lo vai automaticamente aplicar os estilos à página. Entretando, CSS Modules exportam um objeto padão de estilo que mapeia os nomes originais das classes à seus identificadores únicos.

Reviewer: ```suggestion
Astro suporta Módulos CSS usando a convenção de nome `[nome].module.css`. Como qualquer arquivo CSS, importá-lo vai automaticamente aplicar os estilos à página. Entretanto, Módulos CSS exportam um objeto padrão que mapeia os nomes originais de suas classes em identificadores únicos.
```

> CSS Modules ajudam a reforçar o escopo e o isolamento do componente no frontend com os nomes de classes únicos gerados para sua folha de estilos.

Reviewer: ```suggestion
Módulos CSS ajudam a reforçar o escopo e o isolamento de componentes no frontend com nomes de classes únicos gerados para suas folhas de estilos.
```

## Outros Assets

```jsx

> import imgReference from './image.png'; // img === '/src/image.png'
> import svgReference from './image.svg'; // svg === '/src/image.svg'
> import txtReference from './words.txt'; // txt === '/src/words.txt'
> 
> // Esse exemplo usa JSX, mas você pode importar as referência em qualquer framework
> <img src={imgReference} />;

Reviewer: ```suggestion
import referenciaImg from './imagem.png'; // img === '/src/imagem.png'
import referenciaSvg from './imagem.svg'; // svg === '/src/imagem.svg'
import referenciaTxt from './palavras.txt'; // txt === '/src/palavras.txt'

// Esse exemplo usa JSX, mas você pode importar as referências em qualquer framework.
<img src={referenciaImg} />;
```

```

> Todos os outros assets que não foram explicitamente mencionados acima podem ser importados via `import` do ESM e irá retornar a URL de referência à construção final do asset. Isto pode ser útil para referenciar assets que não são JavaScript pela URL, como por exemplo, criar um elemento de imagem com o atributo `src` apontando para aquela imagem.
> 
> Também pode ser útil colocar as imagens no diretório `public/` como explicado na [página de estutura do projeto](/pt-BR/core-concepts/project-structure/).

Reviewer: ```suggestion
Todos os outros assets que não foram explicitamente mencionados acima podem ser importados via `import` do ESM e irão retornar a URL de referência à construção final do asset. Isto pode ser útil para referenciar assets que não são JavaScript pela URL, como por exemplo, criar um elemento `img` com o atributo `src` apontando para aquela imagem.

Também pode ser útil colocar as imagens no diretório `public/` como explicado na [página de estrutura de projetos](/pt-BR/core-concepts/project-structure/).
```

## WASM

```js
// Carrega e inicializa o arquivo WASM requisitado
const wasm = await WebAssembly.instantiateStreaming(fetch('/exemplo.wasm'));
```

> Astro da o suporte para carregamento de arquivos WASM (Web Assembly) diretamente na sua aplicação usando a API [`WebAssembly`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly) do navegador.

Reviewer: ```suggestion
Astro suporta o carregamento de arquivos WASM (Web Assembly) diretamente na sua aplicação usando a API [`WebAssembly`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly) do navegador.
```

> ## Módulos Integrados Node
> 
> Nós recomendamos aos usuários Astro evitar o uso das Módulos integrados do Node.js (`fs`, `path` e etc) sempre quando possível. Astro tem por objetivo ser compatível com múltiplos motores JavaScript no futuro. Isto inclui o [Deno](https://deno.land/) e o [Cloudflare Workers](https://workers.cloudflare.com/) que não possuem suporte aos módulos integrados do Node como o `fs`.

Reviewer: ```suggestion
## Módulos Integrados do Node

Nós recomendamos aos usuários do Astro que evitem o uso de módulos integrados do Node.js (`fs`, `path` e etc) sempre que possível. Astro tem o objetivo de ser compatível com múltiplos motores JavaScript no futuro. Isto inclui o [Deno](https://deno.land/) e o [Cloudflare Workers](https://workers.cloudflare.com/) que não possuem suporte aos módulos integrados do Node como o `fs`.
```

> Nossa missão é prover alternativas Astro para os módulos Node.js mais comuns. Entretando, isto ainda não existe hoje. Então, se você _realmente_ precisa usar estes módulos, nós não queremos barrá-lo. Astro suporta os módulos Node.js usando o novo prefixo `node:` do Node. Se você precisa ser um arquivo, por exemplo, você pode fazê-lo assim:

Reviewer: ```suggestion
Nossa missão é prover alternativas Astro para os módulos Node.js mais comuns. Entretanto, isto estas alternativas ainda não existem hoje. Então, se você _realmente_ precisa utilizar estes módulos, nós não queremos o impedir. Astro suporta os módulos Node.js usando o novo prefixo `node:` do Node. Se você precisa ler um arquivo, por exemplo, você pode fazer assim:
```

```astro
---
// Exemplo: importa o módulo "fs/promises" do Node.js
import fs from 'node:fs/promises';

const url = new URL('../../package.json', import.meta.url);
const json = await fs.readFile(url, 'utf-8');

> const data = JSON.parse(json);
> ---
> 
> <span>Versão: {data.version}</span>

Reviewer: ```suggestion
const dados = JSON.parse(json);
---
<span>Versão: {dados.version}</span>
```
